datalake/backend/processing/check_files_checksums/Dockerfile

```diff
@@ -1,6 +1,8 @@
 FROM ubuntu:20.04
 ENV DEBIAN_FRONTEND noninteractive
-RUN apt update && apt install -y python3 && rm -rf /var/lib/apt/lists/*
+RUN apt-get update \
+    && apt-get install --assume-yes --no-install-recommends python3=3.8.2-0ubuntu2 \
```

Comment:
@l0b0 , are you sure we need to specify apt packages versions here ? The only distribution upgrades are bug fixes. I think this is not necessary.

Reply:
Pro locking versions:

- Makes it less likely to break things arbitrarily. Even OS package patches sometimes sneak in functional changes which impact systems. Reproducibility is a huge win.

Cons:

- Even if we install fixed versions of the top level packages, we might get updated versions of their dependencies. Apt AFAIK does not have a lockfile mechanism like NPM and Poetry, meaning that locking all the package dependencies is a bit more of a chore.
- [Dependabot does not yet patch Dockerfile package installs](https://github.com/dependabot/dependabot-core/issues/2129).

Once Dependabot supports `apt-get` within Dockerfiles that fixes both of the above - we could just lock the package dependencies in the `apt-get` command *once* and Dependabot would take care of keeping them up to date. So on balance I agree for now.